pkg/dev/podmandev/reconcile.go

```diff
@@ -0,0 +1,162 @@
+package podmandev
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/fatih/color"
+
+	devfilev1 "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+
+	"github.com/redhat-developer/odo/pkg/api"
+	"github.com/redhat-developer/odo/pkg/component"
+	"github.com/redhat-developer/odo/pkg/dev"
+	"github.com/redhat-developer/odo/pkg/libdevfile"
+	"github.com/redhat-developer/odo/pkg/log"
+	odocontext "github.com/redhat-developer/odo/pkg/odo/context"
+	"github.com/redhat-developer/odo/pkg/watch"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/klog"
+)
+
+func (o *DevClient) reconcile(
+	ctx context.Context,
+	out io.Writer,
+	errOut io.Writer,
+	options dev.StartOptions,
+	componentStatus *watch.ComponentStatus,
+) error {
+	var (
+		appName       = odocontext.GetApplication(ctx)
+		componentName = odocontext.GetComponentName(ctx)
+		devfileObj    = odocontext.GetDevfileObj(ctx)
+		devfilePath   = odocontext.GetDevfilePath(ctx)
+		path          = filepath.Dir(devfilePath)
+	)
+
+	pod, fwPorts, err := o.deployPod(ctx, options)
+	if err != nil {
+		return err
+	}
+	o.deployedPod = pod
+
+	for _, fwPort := range fwPorts {
+		s := fmt.Sprintf("Forwarding from %s:%d -> %d", fwPort.LocalAddress, fwPort.LocalPort, fwPort.ContainerPort)
+		fmt.Fprintf(out, " -  %s", log.SboldColor(color.FgGreen, s))
+	}
+	err = o.stateClient.SetForwardedPorts(fwPorts)
+	if err != nil {
+		return err
+	}
```

Comment:
Can we move this to after files have been sync'ed and the `run`/`debug` command has been launched? I found it a bit surprising to display the ports before the application is actually started.